### PR TITLE
File blobstore

### PR DESCRIPTION
**Pull request by colmaengus**
Hi Daniel,

I've created an implementation of a file blobstore.
I've not written much python so hopefully it looks okay to you.
I'm planning on adding some tests but if you would take a look in the meanwhile that would be great.

Regards

Colm A

### CODE REVIEW

**Comment by fdr** [comment]
Wow, neat. As it would turn out I'm doing a round of WAL-E work -- it tends to come in fits -- and I'll have a look in a week or two. I'm looking to put out a v0.9.

File: wal_e/blobstore/file/calling_format.py [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+def common_dir_path(args, sep='/'):
+    """ return the highest common directory given a list of files """
+    return os.path.commonprefix(args).rpartition(sep)[0]
+
+
+def epoch_to_iso8601(timestamp):
+    return datetime.utcfromtimestamp(timestamp).isoformat()
+
+
+class FileKey:
```
Comment (fdr) [inline wal_e/blobstore/file/calling_format.py]: For historical reasons, in Python2 use "new-style" classes, which would be: "Filekey(object)" here.

**Comment by fdr** [comment]
Can you make a "blackbox" "config" for this? I think it'd really help your iteration, too.

As you can see, storage backend coverage isn't that great for anything except AWS, but I think it should provide a decent example....it's probably more complex than what yours would have to be.

https://github.com/wal-e/wal-e/blob/master/tests/blackbox.py#L14

**Comment by colmaengus** [comment]
I wasn't clear how the blackbox tests were working so I just added some lower level tests for the file blobstore.
Note that the travis-ci checks are failing due to issue with Azure code. I tried following the suggestions in #201 but it still can't find WindowsAzureMissingResourceError.

**Comment by fdr** [comment]
Try rebasing against the v0.9 branch, I fixed some bitrot there and I think this patch and Google Storage, plus some refreshes on AWS will comprise the v0.9 feature set.

Thanks for the patch.

**Comment by colmaengus** [comment]
Superseded by #204

**Pull request closed without merging**